cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -390,6 +390,38 @@ TEST_P(TestParquetFileFormatScan, ScanRecordBatchReaderWithVirtualColumn) {
 TEST_P(TestParquetFileFormatScan, ScanRecordBatchReaderWithDuplicateColumn) {
   TestScanWithDuplicateColumn();
 }
+TEST_P(TestParquetFileFormatScan, ScanRecordBatchReaderWithFieldPathFilter) {
+  TestScanWithFieldPathFilter();
+}
+TEST_P(TestParquetFileFormatScan, ScanRecordBatchReaderWithFieldPathMultiFileFilter) {
+  compute::Expression filter = greater(field_ref(0), literal(20));
+  auto fields = {field("x", int32()), field("y", int32()), field("z", int32())};
+  auto dataset_schema = schema(fields);
+  SetSchema(fields);
+  SetFilter(filter);
+  auto table = TableFromJSON(dataset_schema,
+                             {
+                                 R"([[10, 20, 30]])",
+                             });
+
+  auto options =
+      checked_pointer_cast<ParquetFileWriteOptions>(format_->DefaultWriteOptions());
+  options->writer_properties = parquet::WriterProperties::Builder()
+                                   .build();
+  options->arrow_writer_properties = parquet::ArrowWriterProperties::Builder()
+                                         .build();
+
+  auto written = WriteToBufferFromTable(dataset_schema, table, options);
+
+  EXPECT_OK_AND_ASSIGN(auto fragment, format_->MakeFragment(FileSource{written}));
+
+  for (auto maybe_batch : PhysicalBatches(fragment)) {
+    ASSERT_OK_AND_ASSIGN(auto batch, maybe_batch);
+    std::cout << "Batch ::" << std::endl;
+    std::cout << batch->ToString() << std::endl;
```

Comment:
We shouldn't print the data, also there doesn't seem to be an assertion in this test?

Reply:
This is part of addressing a review.